core/lib/l1_contract_interface/src/i_executor/structures/commit_batch_info.rs

```diff
@@ -92,24 +147,99 @@ impl<'a> Tokenizable for CommitBatchInfo<'a> {
                 ),
                 // `eventsQueueStateHash`
                 Token::FixedBytes(
-                    self.0
+                    self.l1_batch_with_metadata
                         .metadata
                         .events_queue_commitment
                         .unwrap()
                         .as_bytes()
                         .to_vec(),
                 ),
                 // `systemLogs`
-                Token::Bytes(serialize_commitments(&self.0.header.system_logs)),
-                // `totalL2ToL1Pubdata`
+                Token::Bytes(serialize_commitments(
+                    &self.l1_batch_with_metadata.header.system_logs,
+                )),
+            ]
+        }
+    }
+}
+
+impl<'a> Tokenizable for CommitBatchInfo<'a> {
+    fn from_token(_token: Token) -> Result<Self, Web3ContractError>
+    where
+        Self: Sized,
+    {
+        // Currently there is no need to decode this struct.
+        // We still want to implement `Tokenizable` trait for it, so that *once* it's needed
+        // the implementation is provided here and not in some other inconsistent way.
+        Err(Web3ContractError::Api(Web3ApiError::Decoder(
+            "Not implemented".to_string(),
+        )))
+    }
+
+    fn into_token(self) -> Token {
+        let mut tokens = self.base_tokens();
+
+        let protocol_version = self
```

Comment:
Check here for `IGNORE_1_4_2_UPGRADE_FOR_UPGRADE_TEST`, if it's set then downgrade version from 21 to 20., so upgrade test can pass

Reply:
wdym?

Reply:
I mean upgrade test is failing in CI and I'm suggesting using this kludge (until [this](https://github.com/matter-labs/zksync-era/pull/1262) gets merged) so we can merge PR

Reply:
@koloz193 I will push it